modules/sitemaps/sitemaps.php

```diff
@@ -202,6 +202,9 @@ function jetpack_load_xsl( $type = '' ) {
 
 	$transient_xsl = empty( $type ) ? 'jetpack_sitemap_xsl' : "jetpack_{$type}_sitemap_xsl";
 
+	// Add current language as suffix if WPML is active
+	$transient_xsl = $transient_xsl . wpml_transient_language();
```

Comment:
That's a fine function name, but still would need to put this in the compat file. It's going to get messy to have inline fixes for specific plugins. Just this one isn't terrible, but then Polylang will need something, then something will want something, etc.

Could have a filter for the transient name and likely use existing filters as hooks to call the language switch functions.